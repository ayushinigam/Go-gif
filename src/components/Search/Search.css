.searchInput {
    display: flex; 
    justify-content: center; 
    align-items: center;
    background: #F0F0F0;
    width: 100%;
    padding: 10px; 
    margin: 20px;
    border: 1px solid transparent;
    border-radius: 5px;
    padding: 10px;
    color: #606060;
    font-weight: 500;
}