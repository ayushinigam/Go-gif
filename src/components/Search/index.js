import React, {Component} from 'react';
import PropTypes from 'prop-types'; 
import './Search.css';

class Search extends Component {
    state = {
        value: ''
    }
    onKeyDown = (event) => event.key === 'Enter' && this.updateValue();
    onChange = (event) => this.setState({value: event.target.value});
    updateValue = () => this.props.updateSearchValue(this.state.value);
   render() {       
       return (
           <input type = 'text' 
           onKeyDown = {this.onKeyDown}
           onChange={this.onChange}
           onBlur = {this.updateValue}
           placeholder='Search'
           className="searchInput"
           />
       )
   }
}

Search.defaultProps = {
    updateSearchValue: () => {}
}

Search.propTypes = {
    updateSearchValue: PropTypes.func
}

export default Search