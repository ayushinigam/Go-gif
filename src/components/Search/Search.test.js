import React from 'react';
import renderer from 'react-test-renderer';
import Search from './index';

it('renders correctly', () => {
  const tree = renderer.create(<Search/>).toJSON();
  expect(tree).toMatchSnapshot();
});
