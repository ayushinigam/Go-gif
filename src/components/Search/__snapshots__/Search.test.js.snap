// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<input
  className="searchInput"
  onBlur={[Function]}
  onChange={[Function]}
  onKeyDown={[Function]}
  placeholder="Search"
  type="text"
/>
`;
